model/task_lifecycle.go

```diff
@@ -471,6 +471,9 @@ func MarkEnd(t *task.Task, caller string, finishTime time.Time, detail *apimodel
 		return errors.Wrap(err, "checking for failed tests")
 	}
 	if hasFailedTests {
+		if detailsCopy.Status != evergreen.TaskFailed {
```

Comment:
nit: can we combine into one if statement? We don't need to do 477 if detailsCopy.Status = evergreen.TaskFailed already anyway, since it's redundant

Reply:
good point, I'll combine.

Reply:
good point, I'll combine.

Reply:
I put it in an else.